### PR TITLE
e2e document part3

### PR DESCRIPTION
**Pull request by itsmekumari**
Hi @tivv @rmstar Added e2e document part3. Please review.

### CODE REVIEW

File: Addnewplugin.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# New Plugin in Google-cloud
+
+## Add New Tests for New Plugin
+
+1. All the details will be under the e2e-test folder structure.
+
+**Add Feature:**
+>Feature file contains high level description of the Test Scenarios in simple language.
```
Comment (rmstar) [inline Addnewplugin.md]: link to documentation about features

Comment (itsmekumari) [reply Addnewplugin.md]: Done.

File: Addnewplugin.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# New Plugin in Google-cloud
```
Comment (itsankit-google) [inline Addnewplugin.md]: Please don't mention google-cloud here, documentation should be generic.

Comment (itsmekumari) [reply Addnewplugin.md]: Done.

**Review by itsankit-google — APPROVED** [review]

**Comment by itsankit-google** [comment]
Please squash the commits.

File: Addnewplugin.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+1. All the automation code should strictly be under the src/e2e-test/.
+
+**Add Feature:**
+>Refer [documentation](https://cucumber.io/docs/gherkin/reference/) about features.
+
+1. All the Feature files are at path src/e2e-test/features/.
+2. Plugin specific code is under their respective package name at src/e2e-test/java/io/cdap/plugin/ with below sub packages:
+
+   a. tests.runner</br>
```
Comment (rmstar) [inline Addnewplugin.md]: please describe what code lives under each package

File: Addnewplugin.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+1. All the automation code should strictly be under the src/e2e-test/.
+
+**Configuration Details required for Setup:**
+
+1. Create a maven profile in the main project for which e2e tests are required, with a dependency of e2e test framework artifact, along with your Runner file inside the failsafe plugin.
+
+NOTE: All the properties/ actions/ locators which are common to plugins, need to be or will be present in the framework code. Plugin specific properties/ actions/ locators should reside in the main project module (for eg: google-cloud).
+
+2. TestRunners and its step definition should also resides in the main project module.
```
Comment (rmstar) [inline Addnewplugin.md]: nit: resides -> reside

File: Addnewplugin.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+3. To maintain the service account, add the configuration for TestRunner.
```
Comment (rmstar) [inline Addnewplugin.md]: What does this mean? - "To maintain the service account, add the configuration for TestRunner.". 
Does the user just need to set GOOGLE_APPLICATION_CREDENTIALS environment variable? Any other configuration??

File: Addnewplugin.md (hunk 3 above)
Comment (itsmekumari) [reply Addnewplugin.md]: Done.

File: Addnewplugin.md (hunk 4 above)
Comment (itsmekumari) [reply Addnewplugin.md]: Done.

File: Addnewplugin.md (hunk 5 above)
Comment (itsmekumari) [reply Addnewplugin.md]: Yes, this is the only path to specify in environment variables, apart from that, everything else is in property files.

File: Addnewplugin.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+1. All the automation code should strictly be under the src/e2e-test/.
+
+**Configuration Details required for Setup:**
+
+1. Create a maven profile in the main project for which e2e tests are required, with a dependency of e2e test framework artifact, along with your Runner file inside the failsafe plugin.
+
+NOTE: All the properties/ actions/ locators which are common to plugins, need to be or will be present in the framework code. Plugin specific properties/ actions/ locators should reside in the main project module (for eg: google-cloud).
+
+2. TestRunners and its step definition should also reside in the main project module.
+3. To mention the service account, add the configuration for TestRunner.
```
Comment (rmstar) [inline Addnewplugin.md]: It's still not clear what configuration needs to be specified.
If you just need to set GOOGLE_APPLICATION_CREDENTIALS environment variable, then you can change this to:
Specify service account credentials by setting environment variable GOOGLE_APPLICATION_CREDENTIALS=*`Path of JSON service key`*

Comment (itsmekumari) [reply Addnewplugin.md]: Modified.

**Review by rmstar — APPROVED** [review]
please squash commits

**Pull request merged**